src/App-Question-2/HomePage.js

```diff
@@ -0,0 +1,73 @@
+
+import React from 'react';
+import {browserHistory} from 'react-router';
+
+export default class HomePage extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      text: {
+        username:"",
+        password:""
+      },
+      credentials : {
+        username:"mayur",
+        password:"mayur"
+      }
+    };
+
+    this.checkAuth = this.checkAuth.bind(this);
+    this.setInput = this.setInput.bind(this);
+  };
+
+  checkAuth(){
+
+    if((this.state.text.username == this.state.credentials.username)&& (this.state.text.password == this.state.credentials.password)){
```

Comment:
can save state in local variables to avoid writing `this.state` everytime

```javascript
let {text, credentials} = this.state 
```

```javascript
+    if((text.username == credentials.username) && (text.password == credentials.password)){
```